pkg/statistics/handle/storage/read.go

```diff
@@ -843,3 +843,35 @@ func StatsMetaByTableIDFromStorage(sctx sessionctx.Context, tableID int64, snaps
 	count = rows[0].GetInt64(2)
 	return
 }
+
+// convertBoundFromBlob reads the bound from blob. The `blob` is read from the `mysql.stats_buckets` table.
+// The `convertBoundFromBlob(convertBoundToBlob(a))` should be equal to `a`.
+// TODO: add a test to make sure that this assumption is correct.
+func convertBoundFromBlob(ctx types.Context, blob types.Datum, tp *types.FieldType) (types.Datum, error) {
+	// For `BIT` type, when converting to `BLOB`, it's formated as an integer (when it's possible). Therefore, we should try to
+	// parse it as an integer first.
+	if tp.GetType() == mysql.TypeBit {
+		var ret types.Datum
+
+		// The implementation of converting BIT to BLOB will try to format it as an integer first. Theoretically, it should
+		// always be able to format the integer because the `BIT` length is limited to 64. Therefore, this err should never
+		// happen.
+		uintValue, err := strconv.ParseInt(string(blob.GetBytes()), 10, 64)
```

Comment:
Another question, are the bucket bounds always a value from the table? 

If not (e.g. it's somehow calculated), it's possible that it's an invalid utf8mb4 string or it's not an integer string. Then this logic is incorrect and I don't know how to do it properly 🤔 .

Reply:
Yes. We can assume it's always a real value from the table.